test/testdata/infer/attached_class.rb

```diff
@@ -0,0 +1,26 @@
+# typed: true
+
+class Model
+  extend T::Sig
+
+  sig {params(args: T.untyped).returns(AttachedClass)}
+  def self.load(args)
+    T.unsafe(nil)
+  end
+end
+
+class A < Model
+end
```

Comment:
Suggestion: deeper hierarchy? Parent / Child / GrandChild
Suggestion: abstract parent + concrete child
Suggestion: an implementation that's not `T.unsafe(nil)` (is this possible? does `new` in the parent work like you might think it does?)

```ruby
# typed: true

class Parent
  def self.make
    new
  end
end

class Child1 < Parent
end
class Child2 < Parent
end

puts Parent.make.class
puts Child1.make.class
puts Child2.make.class
```

output:

```
Parent
Child1
Child2
```

Reply:
Great ideas, I'll add them. Using `new` in the parent should work, so I'll switch to that instead of `T.unsafe(nil)`.

Reply:
So there's a problem with using `new` that I think is unrelated to this change, but using `T.nilable(AttachedClass)` also works just fine. I'll timebox investigating the issue with `new`.